core/src/main/java/hudson/model/Run.java

```diff
@@ -1538,7 +1539,19 @@ public Collection<Fingerprint> getBuildFingerprints() {
      * @since 1.349
      */
     public void writeLogTo(long offset, @NonNull XMLOutput out) throws IOException {
-        getLogText().writeHtmlTo(offset, out.asWriter());
+        long start = offset;
+        if (offset > 0) {
+            try (BufferedInputStream bufferedInputStream = new BufferedInputStream(getLogInputStream())) {
```

Comment:
This is a serious performance regression for Pipeline as discovered in #10515: as of https://github.com/jenkinsci/workflow-job-plugin/pull/27 this method is not intended to scale well (https://github.com/jenkinsci/workflow-job-plugin/pull/27#discussion_r221681551).